scripts/check-markdown-links.py

```diff
@@ -0,0 +1,105 @@
+"""Check that links in markdown files point to reasonable places."""
+
+import os
+import re
+import subprocess
+import sys
+from functools import cache
+from http.client import responses as status_code_names
+from pathlib import Path
+
+import requests
+
+PROJECT_ROOT = Path(__file__).absolute().parent.parent
+assert (PROJECT_ROOT / "README.md").is_file()
+os.chdir(PROJECT_ROOT)
+
+
+def find_markdown_files():
+    output = subprocess.check_output(["git", "ls-files", "*.md"], text=True)
+    return [Path(line) for line in output.splitlines()]
+
+
+def find_links(markdown_file_path):
+    content = markdown_file_path.read_text(encoding="utf-8")
+
+    if markdown_file_path.name == "CHANGELOG.md":
+        # Ignore changelogs of old versions. Editing them doesn't make sense.
+        header_matches = list(re.finditer("^## v", content, flags=re.MULTILINE))
```

Comment:
This must also match `## UNRELEASED`
```suggestion
        header_matches = list(re.finditer("^## ", content, flags=re.MULTILINE))
```